api/base_controller.go

```diff
@@ -273,3 +310,84 @@ func stripCredentials(ctx context.Context, object types.Object) {
 		log.C(ctx).Debugf("Object of type %s with id %s is not secured, so no credentials are cleaned up on response", object.GetType(), object.GetID())
 	}
 }
+
+func (c *BaseController) parseMaxItemsQuery(maxItems string) (int, error) {
+	limit := c.DefaultPageSize
+	var err error
+	if maxItems != "" {
+		limit, err = strconv.Atoi(maxItems)
+		if err != nil {
+			return -1, &util.HTTPError{
+				ErrorType:   "InvalidMaxItems",
+				Description: fmt.Sprintf("max_items should be integer: %v", err),
+				StatusCode:  http.StatusBadRequest,
+			}
+		}
+		if limit < 0 {
+			return -1, &util.HTTPError{
+				ErrorType:   "InvalidMaxItems",
+				Description: fmt.Sprintf("max_items cannot be negative"),
+				StatusCode:  http.StatusBadRequest,
+			}
+		}
+		if limit > c.MaxPageSize {
+			limit = c.MaxPageSize
+		}
+	}
+	return limit, nil
+}
+
+func (c *BaseController) parsePageToken(token string) (string, error) {
+	targetPageSequence := "0"
+	if token != "" {
+		base64DecodedTokenBytes, err := base64.StdEncoding.DecodeString(token)
+		if err != nil {
+			return "", &util.HTTPError{
+				ErrorType:   "TokenInvalid",
```

Comment:
`TokenMissing` with log. Even though 404 is returned you're providing internal information in the body.